### PR TITLE
[Market] add completed_on in storage order for delay payment and fix a potential bug

### PR DESCRIPTION
**Pull request by MyronFanQiu**
need to calcute the piece of each transfer. Since expired_on will be changed, the information of the duration has been lost.

### CODE REVIEW

File: cstrml/tee/src/lib.rs [hunk 1]
```diff
@@ -340,16 +340,17 @@ impl<T: Trait> Module<T> {
...
-                    // 2. Reset `expired_on` for new order
+                    // 1. Reset `expired_on` for new order
                     if sorder.order_status == OrderStatus::Pending {
                         let current_block_numeric = Self::get_current_block_number();
                         // go panic if `current_block_numeric` > `created_on`
                         sorder.expired_on += current_block_numeric - sorder.created_on;
+                        sorder.effected_on = current_block_numeric;
                     }
 
+                    // 2. Change order status to `Success`
```
Comment (MyronFanQiu) [inline cstrml/tee/src/lib.rs:351]: Looks like the original logic won't be excuted.

File: cstrml/market/src/lib.rs [hunk 2]
```diff
@@ -32,6 +32,7 @@ pub struct StorageOrder<AccountId> {
     pub file_identifier: MerkleRoot,
     pub file_size: u64,
     pub created_on: BlockNumber,
+    pub effected_on: BlockNumber,
```
Comment (zikunfan) [inline cstrml/market/src/lib.rs:35]: change to ’completed_on‘

**Comment by zikunfan** [comment]
Add tests for this solved bug😄

File: cstrml/market/src/lib.rs (hunk 2 above)
Comment (MyronFanQiu) [reply cstrml/market/src/lib.rs]: Done

File: cstrml/tee/src/tests.rs [hunk 3]
```diff
@@ -225,6 +227,7 @@ fn test_for_report_works_success() {
         assert_eq!(Tee::used(), 402868224);
         assert_eq!(Market::storage_orders(Hash::repeat_byte(1)).unwrap_or_default().order_status,
                    OrderStatus::Success);
+        assert_eq!(Market::storage_orders(Hash::repeat_byte(1)).unwrap_or_default().expired_on, 303);
```
Comment (MyronFanQiu) [inline cstrml/tee/src/tests.rs:230]: added test for this bug

**Pull request merged**